### PR TITLE
`jellyfish-wallet-encrypted`

### LINKED ISSUE #133: jellyfish-wallet-mnemonic to support encrypt and decrypt
#### What would you like to be added:

1. `jellyfish-wallet-mnemonic` with `encrypt` and `decrypt` function extended above 24 word.
2. 24 word must unlock the wallet, this just encrypt the 24 words on rest.

#### Why is this needed:

The 24 words mnemonic seed should serve as a cold storage of your HD seed.
However the 24 words should not sit in your device (phone/browser/desktop) unencrypted as rest.
Encryption can be done with biometric authentication provided by device, `jellyfish-wallet-mnemonic` just need to provide the interface to do it.

Reference example: https://github.com/bitcoin/bips/blob/master/bip-0038.mediawiki

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- sub package, library to encrypt mnemonic seed and store
- any private key usage require caller to decrypt storage using a passphrase

#### Which issue(s) does this PR fixes?:

Fixes #133

#### Additional comments?:
wait for dependencies merged, will ease the reading
trimmed down changes into series of PRs
#331 (merged) -> #369 (merged) -> #371 then this (put this into draft first)

### CODE REVIEW

File: packages/jellyfish-wallet-encrypted/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "build": "tsc"
+  },
+  "dependencies": {
+    "@defichain/jellyfish-crypto": "0.0.0",
+    "@defichain/jellyfish-wallet": "0.0.0",
+    "@defichain/jellyfish-wallet-mnemonic": "0.0.0",
+    "bignumber.js": "^9.0.1",
+    "bs58": "^4.0.1",
+    "scryptsy": "^2.1.0",
+    "tiny-secp256k1": "^2.0.0"
```
Comment (ivan-zynesis) [inline packages/jellyfish-wallet-encrypted/package.json:44]: @fuxingloh any idea what's wrong here? The GH action test cannot import this but it works in my machine, `npm run all` passed

File: packages/jellyfish-crypto/__tests__/aes256.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Aes256 } from '../src'
```
Comment (ivan-zynesis) [inline packages/jellyfish-crypto/__tests__/aes256.test.ts:1]: changes inherited from #331

File: packages/jellyfish-crypto/src/aes256.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/**
```
Comment (ivan-zynesis) [inline packages/jellyfish-crypto/src/aes256.ts:1]: changes inherited from #331

File: packages/jellyfish-wallet-encrypted/package.json (hunk 1 above)
Comment (ivan-zynesis) [reply packages/jellyfish-wallet-encrypted/package.json]: Fixed. Should be messed up npm cached hash tree due to accidentally ran npm i using older npm ver. Delete node_module, revert package-lock, force clean npm cache and reinstall fix it.

File: packages/jellyfish-wallet-encrypted/example/demo.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import { Transaction } from '@defichain/jellyfish-transaction/dist'
```
Comment (ivan-zynesis) [inline packages/jellyfish-wallet-encrypted/example/demo.ts:1]: From user perspective.

File: package-lock.json [hunk 5]
```diff
@@ -14505,8 +14526,7 @@
     "node_modules/buffer-xor": {
       "version": "1.0.3",
       "resolved": "https://registry.npmjs.org/buffer-xor/-/buffer-xor-1.0.3.tgz",
-      "integrity": "sha1-JuYe0UIvtw3ULm42cp7VHYVf6Nk=",
-      "dev": true
+      "integrity": "sha1-JuYe0UIvtw3ULm42cp7VHYVf6Nk="
```
Comment (ivan-zynesis) [inline package-lock.json:14529]: Few changed from dev-dep to normal dependencies, should be due to usage of `browserify-aes`

**Comment by fuxingloh** [comment]
I am reviewing this, give me some time to understand it's integrative behavior

**Comment by fuxingloh** [comment]
Closed in favor of #480

**Pull request closed without merging**